version: '3.8'

services:
  app:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: symfony_app
    volumes:
      - .:/var/www/html
      - ./var/log:/var/www/html/var/log
    working_dir: /var/www/html
    depends_on:
      - db
    environment:
      DATABASE_URL: mysql://symfony:symfony@db:3306/symfony

  webserver:
    image: nginx:alpine
    container_name: symfony_webserver
    ports:
      - "8080:80"
    volumes:
      - ./nginx.conf:/etc/nginx/conf.d/default.conf
      - .:/var/www/html
    depends_on:
      - app

  db:
    image: mysql:8.0
    volumes:
      - ./mysql:/var/lib/mysql
    container_name: symfony_db
    environment:
      MYSQL_ROOT_PASSWORD: root
      MYSQL_DATABASE: symfony
      MYSQL_USER: symfony
      MYSQL_PASSWORD: symfony
    ports:
      - "3307:3306"
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      interval: 10s
      retries: 5

networks:
  default:
    driver: bridge